LiteFarmOrg/LiteFarm
# LF-2643: When a task is assigned to a user without a wage prompt to provide a wage

## Description
When a user without a wage is selected as task's assignee, the additional section shows up so that the user can provide a wage.
![Screenshot 2023-01-25 at 11 38 29 PM](https://user-images.githubusercontent.com/33141219/214781517-d783a448-c5d5-4492-98ea-3b92ea98ef57.png)

Jira link: https://lite-farm.atlassian.net/browse/LF-2643

- This section should be shown if these condition below are met:
   - the user is admin
   - the assignee does not have a wage
   - the task does not have `wage_at_moment`
- The hourly wage input should be displayed only if "Yes, set hourly wage" or "Yes, just for this task" option is selected.

## What does this PR do?
- add `wage_do_not_ask_again` column to `userFarm` table
- create APIs
   - `/user_farm/wage_do_not_ask_again/farm/:farm_id/user/:user_id`
   - `/task/patch_wage/:task_id`
- modify styles of the info icon in `Tooltip` component (it was being affected by a className of its parent component and affecting another component)
- add the hourly wage section

## How Has This Been Tested?
Login as admin, go to `/tasks` and:
1.
- click assignee of a task that has `wage_at_moment`. The wage section is hidden regardless of the selected worker
- click assignee of a task that does not have `wage_at_moment`
  - assignee input
    - select "Unassigned" -> the wage section is hidden
    - select a worker with wage -> the wage section is hidden
    - select a worker without wage
       - the wage section is shown.
         - select Yes option. The wage input is shown. An error message is shown if `number < 0 || 999,999,999 < number`
            - "yes, set hourly wage" and click "Update" -> `wage` in `userFarm` table(DB) is updated with the wage inputted
            - "yes, just for this task" and click "Update" -> `wage_at_moment` in `task` table(DB) is updated with the wage inputted
            - "no" and click "Update" -> nothing happens except for task assigning
            - "no, do not ask again" -> `wage_do_not_ask_again` in `userFarm` table(DB) is updated to `True`
    - select a worker that has had “Don’t ask again” set is chosen -> the wage section is hidden

2.
- click a task that has `wage_at_moment`. Click edit icon for the assignee input. The wage section is hidden regardless of the selected worker
- click a task that does not have `wage_at_moment`. The same behaviours are above is seen.

**When logged in as farm worker, the wage section should not be shown.** 

## Note
Task assigning should work as before. (functionality was not updated)
There are two existing bugs that will not be addressed in this PR.
- [LF-2932](https://lite-farm.atlassian.net/browse/LF-2932) - Multiple tasks cannot be unassigned at once
- [LF-2935](https://lite-farm.atlassian.net/browse/LF-2935) - Completed task's assignee is clickable although it cannot be updated

[LF-2932]: https://lite-farm.atlassian.net/browse/LF-2932?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
